### PR TITLE
fix: refresh colorscheme after apply diff

### PR DESCRIPTION
**Pull request by PJalv**
Added line to reapply colorscheme, diff highlights were not showing after applying.
This is the case for me since I have dynamic colorscheme based on filetype.
Before:
image
After:
image

### CODE REVIEW

File: lua/avante/sidebar.lua [hunk 1]
```diff
@@ -291,6 +291,7 @@ function Sidebar:apply()
     Diff.add_visited_buffer(self.code.bufnr)
     Diff.process(self.code.bufnr)
     api.nvim_win_set_cursor(self.code.winid, { 1, 0 })
+    vim.cmd.colorscheme(vim.g.colors_name)
```
Comment (aarnphm) [inline lua/avante/sidebar.lua:294]: This seems like a hack, we should investigate further.

afaik doing this in a plugin is often frowned upon.

**Review by aarnphm — CHANGES_REQUESTED** [review]
check comment.

**Comment by aarnphm** [comment]
I will include this in my refactor of diff highlights and will keep this in mind.

**Comment by aarnphm** [comment]
Can you provide a small reproducible so that I can test with this?

**Comment by PJalv** [comment]
Sure, here is the autocmd I have been using. 
In hindsight my suggestion seems like a hacky way to fix. Will look further.
```lua
vim.api.nvim_create_autocmd({ "BufEnter", "BufRead", "BufWinEnter", "InsertLeave" }, {
  callback = function()
    if vim.bo.filetype == "zig" or vim.bo.filetype == "c" or vim.bo.filetype == "cpp" then
      vim.cmd.colorscheme("gruvbox")
      require("lualine").setup({ options = { theme = "gruvbox" } })
    else
      vim.cmd.colorscheme("tokyonight")
      require("lualine").setup({ options = { theme = "tokyonight" } })
    end
  end,
})
```

**Comment by aarnphm** [comment]
closed due to #247

**Pull request closed without merging**